### PR TITLE
Show extra calculations

### PR DESCRIPTION
**Pull request by tomasmcm**
- adds functions to calculate dummy extra data
- displays extra data in place

TODO:
- [ ] ~add real formulas to calculate CO2, calories and ticket price~ Added card to trello

Needs discussion:
- [ ] ~should we add CO2, calories and ticket price somewhere else? (trip summary row?)~ Added card to trello

### CODE REVIEW

File: app/util/legUtils.js [hunk 1]
```diff
@@ -52,6 +53,13 @@ export const LegMode = {
   BicycleWalk: 'BICYCLE_WALK',
   CityBike: 'CITYBIKE',
   Walk: 'WALK',
+  Bus: 'BUS',
+  Tram: 'TRAM',
+  Ferry: 'FERRY',
+  Rail: 'RAIL',
+  Subway: 'SUBWAY',
+  Airplane: 'Airplane',
```
Comment (ruimagalhaes) [inline app/util/legUtils.js]: @tomasmcm shouldn't we add this in uppercase too?

File: app/util/legUtils.js [hunk 2]
```diff
@@ -65,17 +73,13 @@ export const getLegMode = legOrMode => {
...
-    case LegMode.CityBike:
-      return LegMode.CityBike;
-    case LegMode.Walk:
-      return LegMode.Walk;
-    default:
-      return undefined;
+  
+  const _legMode = (mode || '').toUpperCase();
+
+  if (values(LegMode).indexOf(_legMode) > -1) {
```
Comment (ruimagalhaes) [inline app/util/legUtils.js]: @tomasmcm can we use `find` to simplify?

`return values(LegMode).find(m => m === _legMode)`

File: app/util/legUtils.js [hunk 3]
```diff
@@ -263,3 +273,62 @@ export const getZones = legs => {
   }
   return Object.keys(zones).sort();
 };
+
```
Comment (ruimagalhaes) [inline app/util/legUtils.js:272]: @tomasmcm these functions will be the ones that will be changed once we have the final formulas right?

File: app/util/legUtils.js (hunk 1 above)
Comment (tomasmcm) [reply app/util/legUtils.js]: 🤦‍♂️

File: app/util/legUtils.js (hunk 2 above)
Comment (tomasmcm) [reply app/util/legUtils.js]: 🙌

File: app/util/legUtils.js (hunk 3 above)
Comment (tomasmcm) [reply app/util/legUtils.js:272]: should only have to change `addExtraCalcsToLeg`

File: app/util/legUtils.js [hunk 4]
```diff
@@ -263,3 +269,62 @@ export const getZones = legs => {
...
+*/
+
+export const co2ToString = co2 => {
+  if (!co2) return false;
+
+  return `${parseInt(co2, 10)} kgCO2`;
+};
+
+export const getTotalWalkingCalories = (itinerary) => {
+  let walkingLegs = itinerary.legs.filter(isWalkingLeg);
```
Comment (ruimagalhaes) [inline app/util/legUtils.js]: @tomasmcm let me suggest a more functional approach:

export const getTotalWalkingCalories = (itinerary) => itinerary.legs.filter(isWalkingLeg).map(l => l.calories).reduce((x, y) => (x || 0) + (y || 0), 0)

File: app/util/legUtils.js (hunk 2 above)
Comment (tomasmcm) [reply app/util/legUtils.js]: @ruimagalhaes I'm used to writing code that has to work on IE, so I always fallback to methods that are older. But on this project I think they dropped IE a long time ago 😉

**Review by ruimagalhaes — APPROVED** [review]

**Pull request merged**